app/summary/directives/instructorSummary/scheduledCourses/scheduledCourses.html

```diff
@@ -14,20 +12,47 @@ <h5>Scheduled Courses {{ year.yearToAcademicYear() }}</h5>
 			<div class="scheduled-courses-table__header-cell">Teaching Assistant(s)</div>
 		</div>
 
-		<div class="scheduled-courses-table__content" ng-repeat="term in state.scheduledCourses.terms">
+		<div class="scheduled-courses-table__content" ng-repeat="termCode in termCodes">
 			<div class="scheduled-courses-table_row">
-				<div class="scheduled-courses-table__term">{{ term.getTermCodeDisplayName() }}
-					<div ng-if="isCurrentTerm(term)" class="scheduled-courses-table__green-label">Current Quarter</div>
+				<div class="scheduled-courses-table__term">{{ termCode.getTermCodeDisplayName() }}
+					<div ng-if="isCurrentTerm(termCode)" class="scheduled-courses-table__green-label">Current Quarter</div>
 				</div>
 			</div>
 
-			<div ng-repeat="course in state.scheduledCourses.list[term]">
-				<div ng-repeat="meeting in course.meetings">
-					<div
-					class="scheduled-courses-table__row"
-					ng-class="{'scheduled-courses-table__row--emphasis': meeting.activityType == 'Lecture'}">
+			<div ng-repeat="instructorAssignment in instructorAssignments" ng-if="instructorAssignment.termCode == termCode">
+				<div>
+					<div class="scheduled-courses-table__row scheduled-courses-table__row--emphasis">
+						<div class="scheduled-courses-table__cell">
+							{{ instructorAssignment.description }}
+						</div>
+						<div class="scheduled-courses-table__cell">
+							{{ instructorAssignment.meetings[0].activityType }}
```

Comment:
What if there is no instructorAssignment.meetings[0]?

Reply:
If that instructorAssignment has no meeting, then the activity/time/location columns will be empty (the assignment description will still display, but this code will handle non-course based assignments, and partially scheduled course assignments.